library/Vanilla/Contracts/Web/LegacyViewHandlerInterface.php

```diff
@@ -0,0 +1,21 @@
+<?php
+/**
+ * @author Adam Charron <adam.c@vanillaforums.com>
+ * @copyright 2009-2019 Vanilla Forums Inc.
+ * @license GPL-2.0-only
+ */
+
+namespace Vanilla\Contracts\Web;
+
+/**
+ * An interface for a view handler for Gdn_Controller.
+ */
+interface LegacyViewHandlerInterface {
+    /**
+     * Render the given view.
+     *
+     * @param string $path The path to the view's file.
+     * @param \Gdn_Controller $controller The controller that is rendering the view.
+     */
+    public function render($path, $controller);
```

Comment:
I am not sure what is this interface you need for.
But in my opinion `render()` should not get `controller` as an argument but `data`.

Reply:
`Gdn_Controller` relies on its view handlers having a specific signature for their `render` methods [[1]](https://github.com/vanilla/vanilla/blob/d54d29e307000cbcce3b8a232f72df9dc4f2828b/library/core/class.controller.php#L763) [[2]](https://github.com/vanilla/vanilla/blob/d54d29e307000cbcce3b8a232f72df9dc4f2828b/library/core/class.controller.php#L2036). The interface is to ensure the legacy view handlers obey the same contract.

Only passing in data would definitely be the way to go, if we were writing a new system, but the Twig view handler has to be backwards-compatible to work with `Gdn_Controller`. I think we're stuck doing it the "legacy" way.